sstmac/hardware/snappr/snappr_outport.cc

```diff
@@ -185,7 +187,7 @@ SnapprOutPort::scheduleArbitration()
 }
 
 void
-SnapprOutPort::deadlockCheck(int vl)
+SnapprOutPort::deadlockCheck(SSTMAC_MAYBE_UNUSED int vl)
```

Comment:
This was creating warnings for me with GCC 7

Reply:
Which warnings exactly? I tested with gcc 5.4.0 and 9.2. Are you still getting warnings? g++ has macro expansion parsing issues, that are not present in gcc.  I don't understand the full details, but that code wouldn't warn if this was a C file, but due to bugs / weirdness in the order things happen in g++ you get the warning. I silenced it by just getting ride of -Wattribute when the compiler is g++.  I don't have a great work around.  If you remove the MAYBE_UNUSED and then try to compile with -Wall you get unused-param warnings and I didn't want to have to wrap the name in ifdefs or define the functions twice.

Reply:
I got these previously on devel. They are gone now. Not sure what you changed? But I guess they are fixed.
````
/home/jjwilke/Programs/sst-macro/qos-snappr-rebase/sumi/gatherv.h:79:27: warning: ‘unused’ attribute ignored [-Wattributes]
   SSTMAC_MAYBE_UNUSED int root_;
````